### PR TITLE
Multipage hooks

### LINKED ISSUE #3982: Add hooks for multipage
We should add hooks where we use the multipage function.
Adding these hooks allows to edit the page link including extra query parameters.

### PR DESCRIPTION
**Pull request by chack1172**
Resolves #3982 

I added the 2 hooks in forumdiplay and showthread. Should I add it before all multipage functions?

### CODE REVIEW

**Comment by JordanMussi** [comment]
I think it’s better to add a hook in the multipage function itself https://github.com/mybb/mybb/blob/127418efedffb189151edc1a9a065679a9811e94/inc/functions.php#L1032

Plugins can then use `THIS_SCRIPT` and other variable to determine what page it is being used on. That way the hook is easily available everywhere pagination is used.

**Comment by euantorano** [comment]
I would agree for definite. Most plugins trying to change any one multi page link would likely want to change them all.

**Comment by chack1172** [comment]
You are right, I didn't add it in the function because it is also called in breadcrumb.
Also they can't access the function parameters with the global keyword.

**Comment by euantorano** [comment]
You can always pass the function parameters to the hook, as an array:

```php
$hook_parameters = array(
    'count' => &$count,
    'perpage' => &$perpage,
    'page' => &$page,
    'url' => &$url,
    'breadcrumb' => &$breadcrumb,
);
```

**Comment by chack1172** [comment]
Ok, I pushed the new edits

File: inc/functions.php [hunk 1]
```diff
@@ -1031,13 +1031,22 @@ function redirect($url, $message="", $title="", $force_redirect=false)
...
 		return '';
 	}
 
+	$args = array(
+		'count' => &$count,
+		'perpage' => &$perpage,
+		'page' => &$page,
+		'url' => &$url,
+		'breadcrumb' => &$breadcrumb,
+	);
```
Comment (Sama34) [inline inc/functions.php:1047]: Maybe one new line here? I can't remember if that is in the coding standards.

Comment (euantorano) [reply inc/functions.php:1047]: Yeah, that's not defined in the coding standards, I'm happy without one really.

**Pull request merged**